policy/policy.go

```diff
@@ -27,91 +24,31 @@ func (p *LoggingPolicy) Do(req *azcorePolicy.Request) (*http.Response, error) {
 	startTime := time.Now()
 	resp, err := req.Next()
 
-	method := req.Raw().Method
 	parsedURL, parseErr := url.Parse(req.Raw().URL.String())
 	if parseErr != nil {
 		p.logger.With(
+			"code", "na",
 			"component", "client",
-			"method", "ERROR",
-			"service", "ERROR",
+			"time_ms", "na",
+			"method", "na",
+			"service", "na",
 			"url", req.Raw().URL.String(),
 			"error", parseErr.Error(),
-		).Error("Error parsing request URL: ", parseErr)
-		return nil, parseErr
-	}
-
-	// Example URLs: "https://management.azure.com/subscriptions/{sub_id}/resourcegroups?api-version={version}"
-	// https://management.azure.com/subscriptions/{sub_id}/resourceGroups/{rg_name}/providers/Microsoft.Storage/storageAccounts/{sa_name}?api-version={version}
-	trimmedURL := trimURL(*parsedURL)
-	splitPath := strings.Split(trimmedURL, "/")
-	var resourceType string
-
-	// Find the index of element that contains "api-version" in the URL
-	apiVersionIndex := -1
-	for i, segment := range splitPath {
-		if strings.Contains(segment, "api-version") {
-			apiVersionIndex = i
-			break
-		}
-	}
-
-	if apiVersionIndex > 0 {
-		// The resource type is the segment right before "api-version"
-		resourceType = splitPath[apiVersionIndex]
-		for _, rt := range resourceTypes {
-			if strings.Contains(splitPath[apiVersionIndex-1], rt) {
-				resourceType = rt
-				break
-			}
-		}
-
-		if method == "GET" {
-			if strings.ContainsAny(resourceType, "?/") {
-				// No resource name is present after resource type, must be a LIST operation
-				index := strings.IndexAny(resourceType, "?/")
-				resourceType = resourceType[:index] + " - LIST"
-			} else {
-				resourceType = resourceType + " - READ"
-			}
-		}
-
-	} else {
-		resourceType = trimmedURL
-	}
-
-	// REST VERB + Resource Type
-	methodInfo := method + " " + resourceType
-
-	if err != nil {
-		p.logger.With(
-			"component", "client",
-			"method", methodInfo,
-			"service", parsedURL.Host,
-			"url", trimmedURL,
-			"error", err.Error(),
-		).Error("Error finishing request: ", err)
-		return nil, err
-	}
-
-	// Time is in ms
-	latency := time.Since(startTime).Milliseconds()
-
-	logEntry := p.logger.With(
-		"code", resp.StatusCode,
-		"component", "client",
-		"time_ms", latency,
-		"method", methodInfo,
-		"service", parsedURL.Host,
-		"url", trimmedURL,
-	)
-
-	// separate check here b/c previous error check only checks if there was an error in the req
-	if 200 <= resp.StatusCode && resp.StatusCode < 300 {
-		logEntry.Info("finished call")
+		).Error("error parsing request URL")
 	} else {
-		logEntry.With("error", resp.Status).Error("finished call")
+		// Example URLs: "https://management.azure.com/subscriptions/{sub_id}/resourcegroups?api-version={version}"
+		// https://management.azure.com/subscriptions/{sub_id}/resourceGroups/{rg_name}/providers/Microsoft.Storage/storageAccounts/{sa_name}?api-version={version}
+		trimmedURL := trimURL(*parsedURL)
+		req.Raw().URL.Path = trimmedURL
```

Comment:
Will this affect other policies in the chain? We modified the req which is a pointer.

Reply:
We have no other policies in the chain at the moment, so it is currently not an issue. I have added a URL element to the LogRequestParams struct to avoid modifying the req pointer to avoid any issues in future.